Restructure longhorn UI proxy link to contain dynamic namespace

jordojordo opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #10259 
<!-- Define findings related to the feature or bug issue. -->
The Longhorn frontend UI would fail to load as the `namespace` was hard-coded to `longhorn-system`. Although setting this namespace is the [recommended approach](https://longhorn.io/docs/1.5.3/deploy/install/install-with-helm/#installing-longhorn) when installing Longhorn with Helm, users may still determine their own namespace.

### Occurred changes and/or fixed issues
<!-- Include information of the changes, including collateral areas which have been affected by this PR as requirement or for convenience. -->
This adds a computed property to find the Longhorn UI based on the label `app=longhorn-ui` which is defined in the [Longhorn Helm chart](https://github.com/rancher/charts/blob/38aed06f77002690b4857d84e79252bed828dc82/charts/longhorn/103.2.1%2Bup1.5.3/templates/deployment-ui.yaml#L16). 
If the service is found, the `externalLinks` will contain the correct namespace of this service in the `link` property.

### Technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->
I've also added a banner to display an error when no external links have been populated.


### Screenshot/Video

https://github.com/rancher/dashboard/assets/40806497/929e5993-c8c2-40d8-9792-1b96624590d0

nwmac commented on shell/pages/c/_cluster/longhorn/index.vue:
```diff
@@ -19,18 +22,30 @@ export default {
+
+  async mounted() {
+    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
+      await this.$store.dispatch('cluster/findAll', { type: SERVICE });
```
There might be a large number of services, so for performance, we can use the label selector to look just for the longhorn one.

jordojordo replied:
That was my original approach with:
```js
  async mounted() {
    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
      const uiService = await this.$store.dispatch('cluster/findMatching', {
        type:     SERVICE,
        selector: 'app=longhorn-ui'
      });

      if ( !isEmpty(uiService) && uiService.metadata?.namespace ) {
        this.externalLinks = [
          {
            enabled:     true,
            iconSrc:     this.longhornImgSrc,
            label:       'longhorn.overview.linkedList.longhorn.label',
            description: 'longhorn.overview.linkedList.longhorn.description',
            link:        `/k8s/clusters/${ this.currentCluster.id }/api/v1/namespaces/${ uiService.metadata.namespace }/services/http:longhorn-frontend:80/proxy/`
          },
        ];
      }
    }
  },
```

The service is never found even when it exists. If I fetch all of the services with `findAll`, then run `findMatching` this will work but defeats the purpose. Am I using `findMatching` incorrectly?


https://github.com/rancher/dashboard/assets/40806497/1d605948-ab4a-4235-a423-c6e216ebe16f

nwmac replied:
You can supply `force: true` in the options to bypass that behaviour.

nwmac commented:
@jordojordo I had a look and the issue is that findMatching returns an array of matching services, not a single service which your code was looking for.

This worked for me:
```
  async mounted() {
    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
      await this.$store.dispatch('cluster/findAll', { type: SERVICE });

      const uiServices = await this.$store.dispatch('cluster/findMatching', { type: SERVICE, selector: 'app=longhorn-ui' });

      console.error(uiServices);

      if (uiServices && uiServices.length ===1 && uiServices[0].metadata?.namespace) {
        this.externalLinks = [
          {
            enabled:     true,
            iconSrc:     this.longhornImgSrc,
            label:       'longhorn.overview.linkedList.longhorn.label',
            description: 'longhorn.overview.linkedList.longhorn.description',
            link:        `/k8s/clusters/${ this.currentCluster.id }/api/v1/namespaces/${ uiServices[0].metadata.namespace }/services/http:longhorn-frontend:80/proxy/`
          },
        ];
      }
    }
  },
```

You might want to use fetch rather than mounted as with this, the red banner flicks up briefly before the service is found.

I would also suggest we use a vertical `IconMessage` rather than the error banner, eg:

![image](https://github.com/rancher/dashboard/assets/1955897/c20601b5-99ef-4ab5-8afb-cbba6360c229)

Lastly, if we can avoid using lodash that would be great - generally I think we want to try and get that out of the code base, espcially for things like `isEmpty` (code above does not use that)

nwmac reviewed (changes requested):
See comment

jordojordo commented:
I updated this to use the `fetch` method rather than the `mounted` when fetching the services, converted the `externalLinks` into a computed property, and switched the Banner to an IconMessage component with a more specific message. Also updated the tests to fit this new structure.

nwmac commented on shell/pages/c/_cluster/longhorn/index.vue:
```diff
@@ -3,42 +3,59 @@ import { mapGetters } from 'vuex';
 
+  async fetch() {
+    if ( this.$store.getters['cluster/schemaFor'](SERVICE) ) {
+      await this.$store.dispatch('cluster/findAll', { type: SERVICE });
```
We can use `findMatching`, but it was broken - this PR fixes it  https://github.com/rancher/dashboard/pull/10413

jordojordo commented:
Restructured this to use the `findMatching` action.


https://github.com/rancher/dashboard/assets/40806497/73c6ec8d-0e50-408d-a047-064531850ef3

enneitex commented:
Hey,

I know this is a bit late as this has been merged for a long time but this change means that users need mors RBAC than before to access longhorn UI. They need to be able to list services in every namespaces, this was not required before
